pkg/storage/watch_cache_test.go

```diff
@@ -230,6 +231,25 @@ func TestEvents(t *testing.T) {
 	}
 }
 
+func TestWaitUntilFreshAndList(t *testing.T) {
+	store := newWatchCache(3)
+
+	// In background, wait some time and then update the store.
```

Comment:
What does this time.Sleep really buy you? There's still no guaranteed order of what occurs first between adding rv 5 and waiting on rv at least 4.